extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/ConfigurationService.java

```diff
@@ -223,4 +223,36 @@ public int getMaxResults() throws GuacamoleException {
         );
     }
 
+    /**
+     * Returns whether or not LDAP aliases will be dereferenced,
+     * as configured with guacamole.properties.
+     * By default they will never be dereferenced.
+     *
+     * @return
+     *     An integer representing the status of of alias
+     *     dereferencing, as configured in guacamole.properties.
+     *
+     * @throws GuacamoleException
+     *     If guacamole.properties cannot be parsed.
+     */
+    public int getDereferenceAliases() throws GuacamoleException {
+        String derefAliases = environment.getProperty(
+            LDAPGuacamoleProperties.LDAP_DEREFERENCE_ALIASES,
+            "never"
```

Comment:
Perhaps it would be better to implement a `GuacamoleProperty` subclass which actually (and strictly) parses these values? The list of possible values here is begging for an `enum`.

I see `never` is never explicitly handled below, and any mistake in the spelling of the other values will result in the property silently assuming you meant `never`, rather than throwing an explicit exception.

It would be better if incorrect values result in an exception (which will ultimately be logged).